code/modules/station_goals/shield.dm

```diff
@@ -0,0 +1,227 @@
+//Station Shield
+// A chain of satellites encircles the station
+// Satellites be actived to generate a shield that will block unorganic matter from passing it.
+/datum/station_goal/station_shield
+	name = "Station Shield"
+
+/datum/station_goal/station_shield/get_report()
+	return {"The station is located in a zone full of space debris.
+			 We have a prototype shielding system you will deploy to reduce collision related accidents.
+
+			 You can order the satellites and control systems through cargo shuttle.
+			 "}
+
+
+/datum/station_goal/station_shield/on_report()
+	//Unlock 
+	var/datum/supply_pack/P = SSshuttle.supply_packs[/datum/supply_pack/misc/shield_sat]
+	P.special_enabled = TRUE
+
+	P = SSshuttle.supply_packs[/datum/supply_pack/misc/shield_sat_control]
+	P.special_enabled = TRUE
+
+/datum/station_goal/station_shield/check_completion()
+	var/list/x_coords = list()
+	var/list/y_coords = list()
+	var/list/obj/machinery/shield_sat/sats = list()
+	for(var/obj/machinery/shield_sat/S in machines)
+		if(S.z == ZLEVEL_STATION)
+			if(!S.shields.len)
+				return FALSE
+			sats += S
+	sortTim(sats,/proc/cmp_coords_ccw)
+	var/obj/machinery/shield_sat/F = sats[1]
+	for(var/V in 1 to sats.len)
+		x_coords += F.x
+		y_coords += F.y
+		F = F.next
+	var/area_v = 0
+	var/j = sats.len
+	for(var/i in 1 to sats.len)
+		area_v += (x_coords[i]+x_coords[j]) * (y_coords[j]-y_coords[i])
+		j = i
+	area_v /= 2
+	if(area_v < world.maxx * world.maxy * 0.5) //change this to use station_state %
+		return TRUE
+	return FALSE
+
+
+/obj/effect/meteor_shield
+	desc = "Shield that blocks all unorganic matter from passing through."
+	name = "Meteor Shield"
+	icon_state = "shield-red"
+	anchored = 1
+	opacity = 0
+	density = 1
+	unacidable = 1
+
+/obj/effect/meteor_shield/CanPass(atom/mover)
+	if(!isliving(mover))
+		return ..()
+	return TRUE
+
+/obj/effect/meteor_shield/ex_act(power)
+	return
+
+
+/obj/machinery/shield_sat
+	icon = 'icons/obj/objects.dmi'
+	icon_state = "shieldoff"
+	name = "Shield Satelitte"
+	desc = "Eat this space rocks."
+	var/sat_range = 14
+	var/obj/machinery/shield_sat/next
+	var/obj/machinery/shield_sat/prev
+	var/list/shields = list()
+	var/static/id = 0 //So linking these is bit less of a pain.
+	density = 1
+
```

Comment:
No since it's moved to space. I guess i could add more batteries but primary challenge here is actually setting up the whole network.